### DISCUSSION TITLE
How to read power consumption of single motors

### DISCUSSION THREAD

**Discussion by hu-yue** [robotology/community#95] [Q&A] [answered]
Hi,
Is there a way to read/know the power consumption of each motor?
Accessing voltage/current readings?
Thanks!

**Comment by pattacini** [comment]
Hi @hu-yue 

I think you should go as you were suggesting by getting the relevant quantities (voltage and current) by means of the motor interface API.

Be careful, because as far as I remember the current readings must be enabled from within the configuration file, at least for `CAN` robots.

**Comment by hu-yue** [comment]
Hi, I was able to read the currents from the IAmplifierControl interface. At least, I've got some numbers out, I'm not quite sure how to interpret them though. I couldn't find it written in the documentation.

**Comment by hu-yue** [comment]
I think we can close this.
I already know that current readings are in mA.

**Comment by randaz81** [comment, marked as answer]
IAmplifierControl::getCurrent() returns the instantaneous current consumption in mA.
IAmplifierControl::getPowerSupplyVoltage() returns the power supply voltage of the board, expressed in Volts (work in progress)
The power consumption is thus P=V*I (expressed in mW). 
Currently you can assume V=40V (the voltage provided by the robot power supply) for all the ETH boards.
